src/modules/numberToWords/index.ts

```diff
@@ -1,132 +1,69 @@
-import { trim } from "../../helpers";
+import numbersWordList from "./numbersWordList";
 import { removeCommas } from "../commas";
 import addOrdinalSuffix from "../addOrdinalSuffix/addOrdinalSuffix";
 
-// <Reference path="https://fa.wikipedia.org/wiki/۱۰۰۰۰۰۰۰۰۰_(عدد)" />
-
-/**
- *
- * @category Number conversion
- */
 export interface NumberToWordsOptions {
 	ordinal?: boolean;
 }
 
-interface INumberToWord {
-	[key: string]: string;
-}
+type NumberToWordsType = (inputNumber: number | string, options?: NumberToWordsOptions) => string | TypeError;
 
-const config = {
-	scale: ["", "هزار", "میلیون", "میلیارد"],
-};
+const numberToWords: NumberToWordsType = (inputNumber, options) => {
+	const isNumberValid = (n: number) => typeof n === "number" && Number.isSafeInteger(n) && n !== 0;
+	const isNegative = (n: number) => n < 0;
+	const notValidNumberError = () => TypeError("the number must be a safe integer value");
```

Comment:
TypeError message should follow the PersianTools errors rule while you can see in some other files we've added before.